lib/util/discover.js

```diff
@@ -27,7 +26,44 @@ function discover(dir) {
       return allModules;
     }
 
-    function scan(inDir, parentName, parentVersion) {
+    function isEyeglassPlugin(pkg) {
+      return pkg.keywords && pkg.keywords.indexOf("eyeglass-module") >= 0;
+    }
+
+    function resolvedModuleDef(pkg, parent, inDir, parentName, parentVersion) {
+      var packageName = parentName ? pkg.name : "./";
+      var eyeglassExports;
+      if (pkg.eyeglass) {
+        eyeglassExports = path.join(packageName, pkg.eyeglass);
+      } else {
+        eyeglassExports = packageName; // use node default
+      }
+
+      var resolvedPkg = {
+        name: pkg.name,
+        version: pkg.version,
+        main: resolve(eyeglassExports, parent, inDir)
+      };
+
+      if (parentName) {
+        resolvedPkg.origin = {
+          name: parentName,
+          version: parentVersion
+        };
+      }
+
+      return resolvedPkg;
+    }
+
+    function shouldRecurse(depth) {
+      return (depth !== undefined && depth > 0) || depth === undefined;
+    }
+
+    function nextDepth(depth) {
+      return (depth !== undefined && depth > 0) ? depth - 1 : depth;
+    }
+
+    function scan(inDir, parentName, parentVersion, depth) {
```

Comment:
Not sure `depth` is needed. Perhaps we add it back in when we have a use case? Will reduce total code.